src/domUpdates.js

```diff
@@ -1,29 +1,292 @@
+const mainHeader = document.getElementById('mainHeader');
+const currentTripCard = document.getElementById('currentTrip');
+const logoutBtn = document.getElementById('logout');
+const totalSpentYTD = document.getElementById('totalSpentYTD');
 
 // display upon user log on 👇
+export const displayTravelerDashBoard = (traveler, trips) => {
+  displayYearToDateSpent();
+  renderDestinationsGrid();
+  renderCurrentTrip();
+};
+
+export const displayYearToDateSpent = () => {
+  totalSpentYTD.innerText = `Total spent traveling with us this year: $${currentTraveler.spendingYTD}`;
+};
+
+export const renderDestinationsGrid = () => {
+  cardsGrid.classList.add('cards-grid');
+  mainDisplay.addEventListener('click', displayBookTripPage);
+  mainHeader.innerText = 'Choose Destination';
+  cardsGrid.innerHTML = '';
+  destinationsData.forEach((destination) => {
+    cardsGrid.innerHTML += `
+    <section class="card" id=${destination.id}>
+      <div class="destination-card">
+        <img class="destination-image grow" id=${destination.id} name="${destination.destination}" src="${destination.image}" alt="${destination.alt}">
+        <div class="destination-details">
+          <h4 class="city">${destination.destination}</h4>
+          <p class="lodging-cost">Estimated lodging: $${destination.estimatedLodgingCostPerDay} / night</p>
+          <p class="flight-cost">Estimated roundtrip flight: $${destination.estimatedFlightCostPerPerson}</p>
+        </div>
+      </div>
+    </section>
+    `;
+  });
+};
+
+export const renderCurrentTrip = () => {
+  currentTraveler.currentTrip = currentTraveler.upcomingTrips[0];
+  let startDate = dayjs(currentTraveler.currentTrip.date).format('MM/DD/YYYY');
+  let endDate = dayjs(startDate)
+    .add(currentTraveler.currentTrip.duration, 'day')
+    .format('MM/DD/YYYY');
+  currentTripCard.innerHTML += `
+    <h3>Your current trip: <h3>
+    <section class="card current">
+    <div class="destination-card">
+      <img class="destination-image" src="${currentTraveler.currentTrip.tripsDestination.image}" alt="${currentTraveler.currentTrip.tripsDestination.alt}">
+      <div class="destination-details">
+        <h4 class="city">${currentTraveler.currentTrip.tripsDestination.destination}</h4>
+        <p class="lodging-cost">Number of travelers: ${currentTraveler.currentTrip.travelers}</p>
+        <p class="lodging-cost">Travel dates: ${startDate} - ${endDate}</p>
+        <p class="flight-cost">Trip cost: $${currentTraveler.currentTrip.cost}</p>
+      </div>
+    </div>
+  `;
+};
+
+// event handlers and functions dependent user interation👇
+export const renderTripsGrid = (e) => {
+  cardsGrid.classList.add('cards-grid');
+  console.log(currentTraveler);
+  mainDisplay.removeEventListener('click', displayBookTripPage);
+  let tripType = e.target.id;
+  if (
+    e.target.id === 'upcomingTrips' ||
+    e.target.id === 'pastTrips' ||
+    e.target.id === 'pendingTrips'
```

Comment:
Nice work on the tripType functionality!

Reply:
Thanks 😁